vacation-calendar/src/Code.js

```diff
@@ -0,0 +1,130 @@
+
+  // Get the list of users in the Google Group.
+  var users = GroupsApp.getGroupByEmail(GROUP_EMAIL).getUsers();
+
+  // For each user, find events having one or more of the keywords in the event
+  // summary in the specified date range. Import each of those to the team
+  // calendar.
+  var count = 0;
+  users.forEach(function(user) {
+    var username = user.getEmail().split('@')[0];
+    KEYWORDS.forEach(function(keyword) {
+      var events = findEvents(user, keyword, today, maxDate, lastRun);
+      events.forEach(function(event) {
+        event.summary = '[' + username + '] ' + event.summary;
+        event.organizer = {
+          id: TEAM_CALENDAR_ID
+        };
+        event.attendees = [];
+        console.log('Importing: %s', event.summary);
+        try {
+          Calendar.Events.import(event, TEAM_CALENDAR_ID);
+          count++;
+        } catch (e) {
+          console.error('Error attempting to import event: %s. Skipping.',
+              e.toString());
+        }
+      }); // End foreach event.
+    }); // End foreach keyword.
+  }); // End foreach user.
+
+  PropertiesService.getScriptProperties().setProperty('lastRun', today);
+  console.log('Imported ' + count + ' events');
+}
+
+/**
+ * In a given user's calendar, look for occurrences of the given keyword
+ * in events within the specified date range and return any such events
+ * found.
+ * @param {string} user the user's primary email String.
+ * @param {string} keyword the keyword String to look for.
+ * @param {Date} start the starting Date of the range to examine.
+ * @param {Date} end the ending Date of the range to examine.
+ * @param {Date} opt_since a Date indicating the last time this script was run.
+ * @return {object[]} an array of calendar event Objects.
+ */
+function findEvents(user, keyword, start, end, opt_since) {
+  var params = {
+    q: keyword,
+    timeMin: formatDate(start),
+    timeMax: formatDate(end),
+    showDeleted: true
+  };
+  if (opt_since) {
+    // This prevents the script from examining events that have not been
+    // modified since the specified date (that is, the last time the
+    // script was run).
+    params['updatedMin'] = formatDate(opt_since);
+  }
+  var response;
+  try {
+    response = Calendar.Events.list(user.getEmail(), params);
+  } catch (e) {
+    console.error('Error retriving events for %s, %s: %s; skipping',
+        user, keyword, e.toString());
+    return [];
+  }
+  return response.items.filter(function(item) {
+    // Filter out events where the keyword did not appear in the summary
+    // (that is, the keyword appeared in a different field, and are thus
+    // is not likely to be relevant).
+    if (item.summary.toLowerCase().indexOf(keyword) < 0) {
+      return false;
+    }
+    // If the event was created by someone other than the user, only include
+    // it if the user has marked it as 'accepted'.
+    if (item.organizer && item.organizer.email != user) {
```

Comment:
Suggest flattening this a little to reduce nesting.

if (!item.organizer || item.organizer == user) {
  return true; // User is organizer
}
// Not organizer, check that user is attendee and accepted the event
var attendees = item.attendees || [];
return attendees.some(function(attendee) {
  return attendee.self && attendee.responseStatus == 'accepted';
}

Reply:
Good call, done.